clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -230,22 +167,15 @@ public boolean isValid() {
         }
     }
 
-    public synchronized FutureTransactionalResult initializeTransactions() {
+    public synchronized TransactionalRequestResult initializeTransactions() {
         ensureTransactional();
         transitionTo(State.INITIALIZING);
-        setPidAndEpoch(NO_PRODUCER_ID, NO_PRODUCER_EPOCH);
+        setPidAndEpoch(PidAndEpoch.NONE);
         this.sequenceNumbers.clear();
-        if (transactionCoordinator == null)
-            pendingTransactionalRequests.add(findCoordinatorRequest(FindCoordinatorRequest.CoordinatorType.TRANSACTION, transactionalId, false));
```

Comment:
Actually the old `hasPid` would always return false in this method since we reset the PidAndEpoch at the beginning, so we would always get a pid. 

Also, the `transactionCoordinator` check is also redundant since we always check whether there is a valid coordinator before sending requests that need it. 

So this is a good simplificatino.